tests/tests/tier0/bluechi-list-units-on-all-nodes/test_bluechi_list_units_on_all_nodes.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+
+from typing import Dict, Tuple
+
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.test import BluechiTest
+
+
+node_foo_name = "node-foo"
+node_bar_name = "node-bar"
+
+
+def parse_bluechictl_output(output: str) -> Dict[str, Dict[str, Tuple[str, str]]]:
+    line_pat = re.compile(r"""\s*(?P<node_name>[\S]+)\s*\|
+                              \s*(?P<unit_name>[\S]+)\s*\|
+                              \s*(?P<state>[\S]+)\s*\|
+                              \s*(?P<sub_state>[\S]+)\s*""",
+                          re.VERBOSE)
+    result = {}
+    for line in output.splitlines():
+        if line.startswith("NODE ") or line.startswith("===="):
+            # Ignore header lines
+            continue
+        match = line_pat.match(line)
+        if match:
+            node_units = result.get(match.group("node_name"))
+            if not node_units:
+                node_units = {}
+                node_units[match.group("unit_name")] = (match.group("state"), match.group("sub_state"))
+                result[match.group("node_name")] = node_units
+            elif match.group("unit_name") in node_units:
+                raise Exception(f"Error parsing bluechictl list-units output, unit already reported, line: '{line}'")
+            else:
+                node_units[match.group("unit_name")] = (match.group("state"), match.group("sub_state"))
+        else:
+            raise Exception(f"Error parsing bluechictl list-units output, invalid line: '{line}'")
+
+    return result
+
+
+def verify_units(all_units: Dict[str, Tuple[str, str]], output: str, node_name: str):
+    esc_seq = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+    line_pat = re.compile(r"""\s*(?P<unit_name>\S+)
+                              .*loaded
+                              \s+(?P<state>\S+)
+                              \s+(?P<sub_state>\S+)
+                              \s+.*$
+                          """,
+                          re.VERBOSE)
+    for line in output.splitlines():
+        # Some systemctl output contains ANSI sequences, which we need to remove before matching
+        line = esc_seq.sub('', line)
+
+        match = line_pat.match(line)
+        if match:
```

Comment:
Could we use an "early exit" here and get rid of the `else`? I think thats a bit easier to read and removes nesting. For example:
```python
if not match:
    raise Exception(f"Error parsing systemctl list-units output, invalid line: '{line}'")

# ...
```